api/v1alpha1/hypershiftdeployment_types.go

```diff
@@ -107,6 +107,19 @@ type HypershiftDeploymentSpec struct {
 	// if omitted, a default NodePool will be generated
 	// +optional
 	NodePools []*HypershiftNodePools `json:"nodePools,omitempty"`
+
+	// Credentials are ARN's that are used for standing up the resources in the cluster.
+	Credentials *CredentialARNs `json:"credentials,omitempty"`
+}
+
+type CredentialARNs struct {
+	AWS *AWSCredentials `json:"aws,omitempty"`
+}
+
+type AWSCredentials struct {
```

Comment:
I'm not familiar with ARN and this piece of code, I apologize in advance if I ask dumb questions.

When users create a HypershiftDeployment, can they specify these？
And to my understanding, these fields are just used for recording the name of the ARN, right? I want to know what is the difference between this and [hyd.Spec.HostedClusterSpec.Platform.AWS.xxx.Name](https://github.com/openshift/hypershift/blob/main/api/v1alpha1/hostedcluster_types.go#L539-L567)?

BTW, after this PR is merged, are [these](https://github.com/stolostron/hypershift-deployment-controller/blob/main/pkg/controllers/default-resources.go#L102-L104) still needed?